service/src/test/java/bio/terra/profile/service/profile/flight/update/UpdateProfileFlightTest.java

```diff
@@ -0,0 +1,231 @@
+  @Autowired AzureConfiguration azureConfiguration;
+  @MockBean GcpCrlService crlService;
+
+  @MockBean SamService samService;
+  @MockBean AzureService azureService;
+  @MockBean TpsApiDispatch tpsApiDispatch;
+  @MockBean ProfileDao profileDao;
+
+  AuthenticatedUserRequest userRequest =
+      AuthenticatedUserRequest.builder()
+          .setToken("fake-token")
+          .setSubjectId("fake-sub")
+          .setEmail("example@example.com")
+          .build();
+
+  @Test
+  void updateProfileBillingAccountAndDescriptionSuccess() throws InterruptedException {
+    String newBillingAccount = "newBillingAccount";
+    String newDescription = "newDescription";
+
+    var billingCow = mock(CloudBillingClientCow.class);
+    when(crlService.getBillingClientCow(any())).thenReturn(billingCow);
+    var iamPermissionsResponse =
+        TestIamPermissionsResponse.newBuilder()
+            .addAllPermissions(VerifyUserBillingAccountAccessStep.PERMISSIONS_TO_TEST)
+            .build();
+    when(billingCow.testIamPermissions(any())).thenReturn(iamPermissionsResponse);
+    var profile = ProfileFixtures.createGcpBillingProfile("ABCD1234");
+    var updatedProfile =
+        new BillingProfile(
+            profile.id(),
+            profile.displayName(),
+            newDescription,
+            profile.biller(),
+            profile.cloudPlatform(),
+            Optional.of(newBillingAccount),
+            profile.tenantId(),
+            profile.subscriptionId(),
+            profile.managedResourceGroupId(),
+            profile.createdTime(),
+            profile.lastModified(),
+            profile.createdBy());
+
+    when(samService.hasActions(userRequest, SamResourceType.PROFILE, profile.id()))
+        .thenReturn(true);
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(profile); // first call occurs pre-flight
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(updatedProfile); // second call occurs post-update
+    when(profileDao.updateProfile(profile.id(), newDescription, newBillingAccount))
+        .thenReturn(true);
+
+    var result =
+        profileService.updateProfile(
+            profile.id(),
+            new UpdateProfileRequest()
+                .billingAccountId(newBillingAccount)
+                .description(newDescription),
+            userRequest);
+    assertEquals(profile.id(), result.getId());
+    assertEquals(newBillingAccount, result.getBillingAccountId());
+    assertEquals(newDescription, result.getDescription());
+
+    verify(samService)
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_BILLING_ACCOUNT);
+    verify(samService)
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_METADATA);
+    verify(billingCow).testIamPermissions(any());
+  }
+
+  @Test
+  void updateProfileBillingAccountSuccess() throws InterruptedException {
```

Comment:
This test and `updateProfileDescriptionSuccess` are arguably not worth including. We're already testing that the flight is properly constructed in `UpdateProfileFlightUnitTest` depending on the update request body. 

The one benefit IMO is these tests would catch a bug where the overall flight's success becomes dependent on one of the steps that is not required for the flight (e.g. `VerifyUserBillingAccountAccessStep` adds a required value to the working map that won't be present if the flight is only updating the profile description).